src/examples/allocator/ServerAllocator.sol

```diff
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.27;
+
+import { Compact } from "src/types/EIP712Types.sol";
+import { ITheCompact } from "src/interfaces/ITheCompact.sol";
+import { IAllocator } from "src/interfaces/IAllocator.sol";
+import { IServerAllocator } from "src/interfaces/IServerAllocator.sol";
+import { Compact, COMPACT_TYPEHASH } from "src/types/EIP712Types.sol";
+import { Ownable, Ownable2Step } from "lib/openzeppelin-contracts/contracts/access/Ownable2Step.sol";
+import { ECDSA } from "lib/openzeppelin-contracts/contracts/utils/cryptography/ECDSA.sol";
+import { EIP712 } from "lib/openzeppelin-contracts/contracts/utils/cryptography/EIP712.sol";
+import { IERC1271 } from "lib/openzeppelin-contracts/contracts/interfaces/IERC1271.sol";
+
+contract ServerAllocator is Ownable2Step, EIP712, IServerAllocator {
+    using ECDSA for bytes32;
+
+    // keccak256("Attest(address,address,address,uint256,uint256)")
+    bytes4 private constant _ATTEST_SELECTOR = 0x1a808f91;
+
+    // keccak256("RegisterAttest(address signer,bytes32 attestHash,uint256 expiration,uint256 nonce)")
+    bytes32 private constant _ATTEST_TYPE_HASH = 0xaf2dfd3fe08723f490d203be627da2725f4ad38681e455221da2fc1a633bbb18;
+
+    // keccak256("NonceConsumption(address signer,uint256[] nonces,bytes32[] attests)")
+    bytes32 private constant _NONCE_CONSUMPTION_TYPE_HASH = 0xb06793f900067653959d9bc53299ebf6b5aa5cf5f6c1a463305891a3db695f3c;
+
+    address private immutable _COMPACT_CONTRACT;
+
+    mapping(address => uint256) private _signers;
+    address[] private _activeSigners;
+
+    mapping(bytes32 => uint256) private _attestExpirations;
+    mapping(bytes32 => uint256) private _attestCounts;
+    mapping(bytes32 => bool) private _attestSignatures;
+
+    modifier isSigner(address signer_) {
+        if (!_containsSigner(signer_)) {
+            revert InvalidSigner(signer_);
+        }
+        _;
+    }
+
+    constructor(address owner_, address compactContract_) Ownable(owner_) EIP712("Allocator", "1") {
+        _COMPACT_CONTRACT = compactContract_;
+        ITheCompact(_COMPACT_CONTRACT).__registerAllocator(address(this), "");
+    }
+
+    /// @inheritdoc IServerAllocator
+    function addSigner(address signer_) external onlyOwner {
+        if (_containsSigner(signer_)) {
+            return;
```

Comment:
would it make more sense to revert here with a `SignerAlreadyAdded` error?

Reply:
I am personally not a fan of reverting if the expected result was achieved, but that is personal preference.